docs/site/BelongsTo-relation.md

```diff
@@ -43,6 +43,29 @@ related routes, you need to perform the following steps:
 This section describes how to define a `belongsTo` relation at the model level
 using the `@belongsTo` decorator to define the constraining foreign key.
 
+The definition of the `belongsTo` relation is inferred by using the `@belongsTo`
+decorator. The decorator takes in a function resolving the target model class
+constructor and designates the relation type. It also calls `property()` to
+ensure that the decorated property is correctly defined. Usage of belongsTo
+relation is similar to legacy datasource juggler.
+
+The `@belongsTo` decorator takes three parameters :
+
+- `target model class`, is a mandatory field
+- `relation definition`, is an optional field - has three attributes, keyFrom,
+  keyTo, name
+  - `keyFrom` is a property name of the foreign key on the "source" model (for
+    example, `customerId` on `Order` model). `keyFrom` is always set to the
+    `decorated property name`. `keyFrom` attribute has become a dummy parameter
```

Comment:
Sorry that I miss this sentence.  Is it the intended behavior?  If it's a bug, I think we should open a new ticket to address that rather than putting it in the documentation.

Reply:
From what I remember from my spike work, `keyFrom` is still used by inclusion resolvers. It's an optional field though because we can infer it automatically in most cases.

Reply:
@bajtos it is always automatically inferred from the decorated target. The passed in parameter is never used by inclusion resolvers.

Reply:
@dhmlau it is not a bug. Intuitively the keyFrom attribute is not relevant anymore, because the property name is available via the decorator anyway (as the decorated target).

Reply:
we can clean up by removing the keyFrom attribute or make the decorator signature more understandable, later. for now the docs state things as they are.

Reply:
Do you mean to remove `keyFrom` from metadata? I don't think it's a good idea...
Could you remove the "keyFrom attribute has become a dummy parameter" part or reword it? It's confusing.

Reply:
@deepakrkris I think `keyFrom` is still honored, see the test case https://github.com/strongloop/loopback-next/pull/3726/files#diff-6795fbd89427a8971f81ebe5f4487536R16.

And that PR added `keyFrom` field to the metadata.

Reply:
@jannyHou do you have an end to end test to validate that keyFrom is honoured ? because the checks on metadata only proves that what is passed in is available in the metadata.

Reply:
@agnes512 @jannyHou, I think I might be seeing what @deepakrkris described. I've created my own app (also with Customer and Order). Even I put some irrelevant characters in `keyTo` _and_ `keyFrom`, it seems to be working fine. i.e. 
```
  @belongsTo(() => Customer, {keyFrom: 'aaaaaa', keyTo: 'aaaaaa'})
  customer_id: string;
```
I'm not sure if it is because I have the `hasMany` relations defined already and somehow it got magically discovered.

To move this task and PR forward, I'd like to suggest:
- defer the task and include the scope to investigate why the values in `keyFrom` and `keyTo` are not honored. Either it got somehow discovered or it's a bug, we will document this properly. 
- keep this PR and modify as needed.

@strongloop/loopback-maintainers (especially @deepakrkris @agnes512 @jannyHou), what do you think?

Reply:
sure let's address it in a separate story. It shouldn't be a blocker for this PR.

Reply:
@dhmlau , work on accessors seems to be still in progress , so I can remove references to keyFrom and land this PR.

Reply:
@dhmlau
by 'it works fine' you mean post/get requests? what's the value of `keyFrom` and `keyTo` in your case?
 Have you tried inclusion resolver for these relations? cause they are actually using `keyFrom` and `keyTo` (sorry I didn't realize earlier that these fields are also related to other part of features): [belongTo](https://github.com/strongloop/loopback-next/pull/3721/files#diff-ee138af2e31d5dd9d5c43732a35a0d04R52)
[hasMany](https://github.com/strongloop/loopback-next/pull/3595/files#diff-d6e69145b3b8445f547492d818ef8d6fR53)
I haven't tested it out with inclusion resolver what would happen if we put random names here.

Reply:
@jannyHou @dhmlau @deepakrkris I just tested it real quick. 

I don't know if the app works fine with @dhmlau means we can do post/get requests. I tested with inclusion resolver. 
For `belongsTo` relation, if you have a customized name `cust_id`, you have to specify {`name`} field in the second parameter in decorator `@belongsTo`. `keyFrom` is optional but it **cannot be** a random name. Or it couldn't get related models through the relation.
I think the `keyFrom` field is honored from the decorated property name, see [here](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/relations/belongs-to/belongs-to.decorator.ts#L50).

So I think `keyFrom` shouldn't be removed. And we need to make it clear in the docs.

Reply:
@agnes512, yes, and do post and get. I'm using postgresql, not sure if it matters. 
Could you please share your app in github? thanks!

Reply:
I will also rework my [update keyFrom, keyTo for hasMany and hasOne relations](https://github.com/strongloop/loopback-next/pull/3805) PR to make sure that these explanations will be consistent with inclusion resolver.

Reply:
@dhmlau I use my tests in `packages/repository` :joy: cause I can run it on MySQL and MongoDB. I modified these fields to test it around. I had a feeling that post/get may work cause they don't relate to `keyFrom` or `keyTo`. But I am sure that the inclusion resolver will be affected if it has incorrect `keyFrom`/`keyTo`.

Reply:
I think we can explain what these fields represent. And mention that to customize the property name decorated by `@belongsTo`, you will have to specify the `name` field. Just an idea~

Reply:
keyFrom and keyTo are working as expected with the recent published lb4 version. updating the docs as per current status. I will remove the dummy param comment as well.

Reply:
Yeah just make sure that next time you use the latest code when you're testing something via `lb4 app` :smile: